### PR TITLE
Added RTDEClient constructor with vector recipes

### PR DESCRIPTION
**Pull request by dagare**

### CODE REVIEW

**Comment by dagare** [comment]
I will make a new addition to this PR

**Review by fmauch — COMMENTED** [review]
@dagare Thank you for the contribution. Could you please provide some insight why you see this as beneficial?

I'm considering to suggest only allow passing recipes to the `RTDEClient` class as vectors directly, so parsing a file (if any) would have to be done a level higher up.

@RobertWilbrandt what do you think?

**Comment by urrsk** [comment]
I will suggest we add the support first.
When we can consider to deprecate the file dependent constructor later

File: src/rtde/rtde_client.cpp [hunk 1]
```diff
@@ -49,6 +49,21 @@ RTDEClient::RTDEClient(std::string robot_ip, comm::INotifier& notifier, const st
...
 }
 
+RTDEClient::RTDEClient(std::string robot_ip, comm::INotifier& notifier, const std::vector<std::string>& output_recipe,
+                       const std::vector<std::string>& input_recipe, double target_frequency)
+  : stream_(robot_ip, UR_RTDE_PORT)
+  , output_recipe_(output_recipe)
+  , input_recipe_(input_recipe)
+  , parser_(output_recipe_)
+  , prod_(stream_, parser_)
+  , pipeline_(prod_, PIPELINE_NAME, notifier)
```
Comment (urmahp) [inline src/rtde/rtde_client.cpp:59]: I think that producer_fifo_scheduling should be set to true, such that it looks identical to the original constructor. 
```suggestion
  , pipeline_(prod_, PIPELINE_NAME, notifier, true)
```

**Review by urmahp — COMMENTED** [review]
Thank you @dagare for the contribution. 

I have added a small adjustment, but other than that it looks good.

**Comment by urmahp** [comment]
Before merging I suggest that we add checks in the constructor to see if the recipes are empty and whether "timestamp" is part of the output_recipe.

File: include/ur_client_library/rtde/rtde_client.h [hunk 2]
```diff
@@ -105,6 +105,19 @@ class RTDEClient
...
              const std::string& input_recipe_file, double target_frequency = 0.0);
+
+  /*!
+   * \brief Creates a new RTDEClient object, including a used URStream and Pipeline to handle the
+   * communication with the robot.
+   *
+   * \param robot_ip The IP of the robot
+   * \param notifier The notifier to use in the pipeline
+   * \param output_recipe Vector containing the output recipe
+   * \param input_recipe_file Vector containing the input recipe
```
[the remark this answers is not in the document]
Comment (urrsk) [reply include/ur_client_library/rtde/rtde_client.h:116]: Should fix the CI error as well

**Review by urrsk — CHANGES_REQUESTED** [review]

**Comment by urmahp** [comment]
Thank you for the contribution @dagare.

**Pull request merged**